src/bdc/steps/helpers/text_analyzer.py

```diff
@@ -0,0 +1,185 @@
+
+        Args:
+            text (str): The text to be corrected.
+            language (str, optional): The language setting. Defaults to "en".
+
+        Returns:
+            str: The corrected text.
+        """
+        speller = self._get_speller(language)
+        spell_checker = self._get_spell_checker(language)
+
+        if speller is None and spell_checker is None:
+            log.warn(
+                f"Could not find a spell checker or speller for language '{language}'."
+            )
+            return text
+
+        speller_corrected_text = speller(text) if speller is not None else text
+        split_word = speller_corrected_text.split()
+        spell_checker_corrected_text = (
+            " ".join(
+                spell_checker.correction(word)
+                if spell_checker.correction(word) is not None
+                else word
+                for word in speller_corrected_text.split()
+            )
+            if spell_checker is not None
+            else text
+        )
+
+        return spell_checker_corrected_text
+
+    def find_number_of_spelling_errors(self, text, language="en"):
+        """
+        Find the number of spelling errors in the given text using the specified language.
+
+        Args:
+            text (str): The text to be checked.
+            language (str, optional): The language setting. Defaults to "en".
+
+        Returns:
+            int: The number of spelling errors.
+        """
+        return len(self.find_spelling_errors(text, language))
+
+    def find_spelling_errors(self, text, language="en"):
+        """
+        Find the spelling errors in the given text using the specified language.
+
+        Args:
+            text (str): The text to be checked.
+            language (str, optional): The language setting. Defaults to "en".
+
+        Returns:
+            list: A list of spelling errors.
+        """
+        corrected_text = self.correct_text(text, language)
+        differences = self._find_differences(text, corrected_text)
+        return differences
+
+    def find_number_of_grammatical_errors(self, inp_text, language="en"):
+        """
+        Finds the number of grammatical errors in the input text.
+
+        Args:
+            inp_text (str): The input text to analyze for grammatical errors.
+            language (str, optional): The language of the input text. Defaults to "en".
+            max_retries (int, optional): The maximum number of retry attempts. Defaults to 3.
+
+        Returns:
+            int: The number of grammatical errors found in the input text, or None if an error occurs.
+        """
+
+        max_retries = 5  # Maximum number of retries
+        retry_delay = 5  # Initial delay in seconds (5 seconds)
+        if inp_text is None or len(inp_text) == 0:
+            return None
+
+        for attempt in range(max_retries):
+            try:
+                errors = ltp.check(
```

Comment:
I see language_tool_python still being used here, should it still be there?

Reply:
I know it is confusing but previous one was called `language_tool_kit`, but this one is called `pylanguagetool`. They are actually different, since `pylanguagetool` is way faster and does not hold StrongCopyLeft.

Reply:
Yes, by bad, nice work!